src/GitLink/Linker.cs

```diff
@@ -41,22 +41,37 @@ public static bool Link(string pdbPath, LinkOptions options = default(LinkOption
             {
                 sourceFiles = pdb.GetFilesAndChecksums().Keys.ToList();
 
-                repositoryDirectory = GitDirFinder.TreeWalkForGitDir(Path.GetDirectoryName(sourceFiles.First()));
-                if (repositoryDirectory == null)
+                if (options.GitWorkingDirectory != null)
```

Comment:
There are millions of invalid values for a working directory. And most/all of them will produce an error regardless. Why would I special case "" or " "? What caller is likely to pass in " ", and why should I allow that to be ignored instead of raising an error (since it's likely an error on their part)? The only case I can think of is when the input is from an MSBuild property (where whitespace abounds), and in that case, that should be filtered at the MSBuild task level since that's what receives the "dirty" value.